tests/test_workflow_api.py

```diff
@@ -90,3 +93,195 @@ async def test_workflow_delete(
         # TODO add tasks with API and test cascade delete
 
         assert not await db.get(Workflow, wf_id)
+
+
+async def test_get_workflow(
+    db, client, MockCurrentUser, project_factory, task_factory
+):
+    """
+    GIVEN a Workflow in the db
+    WHEN the endpoint to GET a Workflow by its id is called
+    THEN the Workflow is returned
+    """
+    async with MockCurrentUser(persist=True) as user:
+        project = await project_factory(user)
+        p_id = project.id
+        workflow = {
+            "name": "My Workflow",
+            "project_id": p_id,
+            "task_list": [],
+        }
+        res = await client.post(
+            "api/v1/workflow/",
+            json=workflow,
+        )
+        wf_id = res.json()["id"]
+        res = await client.get(f"/api/v1/workflow/{wf_id}")
+
+        assert res.status_code == 200
+        workflow["id"] = wf_id
+        assert res.json() == workflow
+
+
+async def test_post_newtask(
+    db, client, MockCurrentUser, project_factory, task_factory
+):
+    """
+    GIVEN a Workflow with a list of WorkflowTasks
+    WHEN the endpoint to POST a new WorkflowTask inside
+        the Workflow.task_list is called
+    THEN the new WorkflowTask is inserted in Workflow.task_list
+    """
+    async with MockCurrentUser(persist=True) as user:
+        project = await project_factory(user)
+        workflow = {
+            "name": "My Workflow",
+            "project_id": project.id,
+        }
+        res = await client.post(
+            "api/v1/workflow/",
+            json=workflow,
+        )
+        assert res.status_code == 201
+        wf_id = res.json()["id"]
+
+        workflow = await db.get(Workflow, wf_id)
+        t0 = await task_factory()
+        t1 = await task_factory()
+        await workflow.insert_task(t0.id, db=db)
+        await workflow.insert_task(t1.id, db=db)
+        await db.refresh(workflow)
+
+        assert len(workflow.task_list) == 2
+        assert workflow.task_list[0].task == t0
+        assert workflow.task_list[1].task == t1
+
+        t2 = await task_factory()
+        new_task = {
+            "task_id": t2.id,
+        }
+
+        res = await client.post(
+            f"api/v1/workflow/{wf_id}/add-task/",
+            json=new_task,
+        )
+        assert res.status_code == 201
+        debug(res.json())
+        workflow = WorkflowRead(**res.json())
+
+        assert len(workflow.task_list) == 3
+        assert workflow.task_list[0].task == TaskRead(**t0.__dict__)
+        assert workflow.task_list[1].task == TaskRead(**t1.__dict__)
+        assert workflow.task_list[2].task == TaskRead(**t2.__dict__)
```

Comment:
- [ ] use public method `Task.dict()` instead of private  `__dict__`
- [ ] add test in which you insert a task in the middle instead of appending it at the end of the task_list.